app/src/pages/explorer/image/Image.vue

```diff
@@ -166,63 +147,151 @@ export default {
+        return this.refetchApollo()
+      } else {
+        this.searchEnabled = false
+        this.pageSize = newValue.pageSize || this.pageSize
+        return this.refetchApollo()
       }
+    },
+    '$route.query' (newValue, oldValues) {
+      this.pageNumber = parseInt(newValue.page) || 1
+      this.pageSize = parseInt(newValue.size) || 20
+      this.filter = newValue.type || ''
+      this.searchWord = newValue.q || ''
+      this.dispatchSearch()
+    },
+    // Limit page size for now
+    pageSize (newValue, oldValue) {
+      this.checkPageSize(newValue)
+      this.loadPrevNextImage(this.pageNumber)
     }
   },
   methods: {
     loadPrevNextImage (event) {
+      this.pageNumber = event
       if (!this.searchEnabled) {
-        this.pageNumber = event
+        return this.changeRoute(event, this.pageSize)
+      }
+      this.changeRoute(event, this.pageSize, this.filter, this.searchWord)
+    },
+    refetchApollo () {
+      this.error = ''
+      if (this.searchEnabled) {
+        this.$apollo.queries.images.skip = true
+        this.$apollo.queries.searchImages.skip = false
+        this.$apollo.queries.searchImages.refetch()
+      } else {
+        this.$apollo.queries.searchImages.skip = true
+        this.$apollo.queries.images.skip = false
         this.$apollo.queries.images.refetch()
-        return
       }
-      this.pageNumber = event
-      this.$apollo.queries.images.skip = true
-      this.$apollo.queries.searchImages.skip = false
-      this.$apollo.queries.searchImages.refetch()
     },
     submitSearch () {
-      if (!this.searchWord && !this.filter) return
-      return this.dispatchSearch()
+      if (!this.searchWord || !this.filter) {
+        return this.$store.commit('setSnackbar', {
+          message: 'Enter a search term and select a filter type'
+        })
+      }
+      this.pageNumber = 1
+      this.dispatchSearch()
+      this.changeRoute(this.pageNumber, this.pageSize, this.filter, this.searchWord)
     },
-    async cancelSearch () {
-      this.$router.go(this.$router.currentRoute)
+    async resetSearch () {
+      this.$router.replace({ query: null })
+      this.searchImages = []
+      this.renderText = 'Showing all images'
     },
     async dispatchSearch () {
       await this.$store.commit('explorer/setSelectedFacetFilterMaterialsValue',
-        { type: this.filter, value: this.searchWord })
+        { type: this.filter, value: this.searchWord, size: this.pageSize })
+    },
+    changeRoute (page = 1, size = 20, filter = '', query = '') {
+      const curQuery = this.$route.query
+      // Don't push change if same route to avoid error
+      if (curQuery.page === page && curQuery.size === size && curQuery.type === filter && curQuery.q === query) {
+        this.refetchApollo()
+      } else if (filter === '' || query === '') {
+        if (curQuery.page === page && curQuery.size === size) return this.refetchApollo()
+        this.$router.push({ query: { page, size } })
```

Comment:
The reason for the return in line 216 is that if I push a new route when the parameters haven't changed, vue-router throws an error that looks something like 'avoid navigating to the same route' (I'll double check on the wording). Otherwise if the parameters have changed, then I go ahead and push a new route (line 217)